Ginger/Ginger/Variables/DatabaseListViewHelper.cs

```diff
@@ -0,0 +1,386 @@
+            bool IsConnectionSuccessful = TestSingleDatabase(db);
+            if (IsConnectionSuccessful)
+            {
+                Reporter.ToUser(eUserMsgKey.DbConnSucceed, db?.Name);
+            }
+
+            else
+            {
+                Reporter.ToUser(eUserMsgKey.DbConnFailed, db?.Name);
+            }
+
+        }
+
+        public string GetItemTagsField()
+        {
+            return null;
+        }
+
+        public ListItemUniqueIdentifier GetItemUniqueIdentifier(object item)
+        {
+            return null;
+        }
+
+        public List<ListItemOperation> GetListExtraOperations()
+        {
+            return [];
+        }
+
+        public List<ListItemOperation> GetListOperations()
+        {
+            List<ListItemOperation> operationsList = new List<ListItemOperation>();
+
+            ListItemOperation addNew = new ListItemOperation();
+            addNew.SupportedViews = new List<General.eRIPageViewMode>() { General.eRIPageViewMode.Automation, General.eRIPageViewMode.SharedReposiotry, General.eRIPageViewMode.Child, General.eRIPageViewMode.ChildWithSave, General.eRIPageViewMode.Standalone };
+            addNew.AutomationID = "addNew";
+            addNew.ImageType = Amdocs.Ginger.Common.Enums.eImageType.Add;
+            addNew.ToolTip = "Add New Database";
+            addNew.OperationHandler = AddNewHandler;
+            operationsList.Add(addNew);
+
+            ListItemOperation deleteSelected = new ListItemOperation();
+            deleteSelected.SupportedViews = new List<General.eRIPageViewMode>() { General.eRIPageViewMode.Automation, General.eRIPageViewMode.SharedReposiotry, General.eRIPageViewMode.Child, General.eRIPageViewMode.ChildWithSave, General.eRIPageViewMode.Standalone };
+            deleteSelected.AutomationID = "deleteSelected";
+            deleteSelected.ImageType = Amdocs.Ginger.Common.Enums.eImageType.Delete;
+            deleteSelected.ToolTip = "Delete Selected Database (Del)";
+            deleteSelected.OperationHandler = DeleteDBs;
+            operationsList.Add(deleteSelected);
+
+            ListItemOperation testAllDatabases = new ListItemOperation();
+            testAllDatabases.SupportedViews = new List<General.eRIPageViewMode>() { General.eRIPageViewMode.Automation, General.eRIPageViewMode.SharedReposiotry, General.eRIPageViewMode.Child, General.eRIPageViewMode.ChildWithSave, General.eRIPageViewMode.Standalone };
+            testAllDatabases.AutomationID = "testAllDatabases";
+            testAllDatabases.ImageType = Amdocs.Ginger.Common.Enums.eImageType.Run;
+            testAllDatabases.ToolTip = "Test All Databases";
+            testAllDatabases.OperationHandler = TestAllDatabases;
+
+            operationsList.Add(testAllDatabases);
+
+            return operationsList;
+        }
+
+        private void TestAllDatabases(object sender, RoutedEventArgs e)
+        {
+
+            foreach (var database in Databases)
+            {
```

Comment:
can we test them in parallel?

Reply:
I tried parallel for each but the whole app hangs. it instead works better without the parallel for each